esp/src/src-react/components/forms/Fields.tsx

```diff
@@ -109,23 +109,30 @@ const AsyncDropdown: React.FunctionComponent<AsyncDropdownProps> = ({
 
     React.useEffect(() => {
         const item = selOptions?.find(row => row.key === selectedKey) ?? selOptions[0];
+        if (!item) return;
+        if (item?.key === selectedKey) {
+            if (!selectedItem) {
```

Comment:
Think this would read better as:
```
if (item?. Key === selectedKey && !selectedItem) {
...
```

Reply:
This wouldn't be exactly the same. I added a comment for what should be happening.